backend/routes.py

```diff
@@ -0,0 +1,145 @@
+import uuid as u
+import copy
+import backend.pike as pike
+import tests.api_fixtures as api_mocks
+
+
+## Read calls
+# c.iii.1 (?) get available agents
+@pike.api.get("/agents")
+def get_public_agents() -> list[dict]:
+    """
+    Get a list of all public agent types.
+    """
+    return [api_mocks.mock_agent_one_skill, api_mocks.mock_agent_two_skills]
+
+
+# 1.d.  User settings
+@pike.api.get("/user/{user_id}")
+def get_user_info(user_id: str) -> dict:
+    """
+    Provides full info about a user given their user_id.
+    """
+    return api_mocks.mock_user
+
+
+# 1.a.  get my agents
+@pike.api.get("/user/{user_id}/agents")
+def get_user_agents(user_id: u.UUID) -> list[dict]:
+    """
+    Get all agents the user has chosen to enable.
+    """
+    return api_mocks.mock_user["agents"]
+
+
+# 1.b.  get compact chat representations
+@pike.api.get("/user/{user_id}/chats")
+def get_user_chats_compact(user_id: str) -> list[dict]:
+    """
+    Gets a list of chat tags, providing enough information to render the associated
+    chats without messages.
+    """
+    return api_mocks.chat_without_messages, api_mocks.additional_chat_without_messages
+
+
+# 2.a
+@pike.api.get("/user/{user_id}/agent/{agent_id}/chats")
+def get_user_chats_by_agent(user_id: str, agent_id: u.UUID) -> list[dict]:
+    """
+    Return a list of all chats for a specific user which use a specific agent.
+    """
+    return api_mocks.chat_without_messages
+
+
+# 2.b (somwhat) get a chat's full history
+@pike.api.get("/user/{user_id}/chat/{chat_id}")
+def get_user_chat(user_id: str, chat_id: u.UUID) -> dict:
+    """
+    Provides the chat history for user {user_id} and thread {chat_id} in an
+    appropriate format for sending to the frontend.
+    """
+    return api_mocks.chat_with_messages
+
+
+# 2.h Get attachment information.  (Should be implicitly filtered by user as attachments are in user's chats)
+@pike.api.get("/attachment/{attachment_id}")
+def get_attachment(attachment_id: u.UUID) -> str:
+    """
+    Uses an attachment_id to request the data from a specific attachment from the backend.
+    """
+    return api_mocks.attachment_database.values()[0]
```

Comment:
let's return a mock_attachment here, and handle the `.values()[0]` inside the fixture